test/cbmc/proofs/Posix_OtaDeleteTimer/Posix_OtaDeleteTimer_harness.c

```diff
@@ -0,0 +1,38 @@
+/*
+ * AWS IoT Over-the-air Update v3.0.0
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file Posix_OtaDeleteTimer_harness.c
+ * @brief Implements the proof harness for Posix_OtaDeleteTimer function.
+ */
+/*  FreeRTOS includes for OTA library. */
+#include "ota_os_posix.h"
+
+void Posix_OtaDeleteTimer_harness()
+{
+    OtaTimerId_t otaTimer;
+
+    /* The valid values for OtaTimerId_t enum are [0,2). */
+    __CPROVER_assume( otaTimer >= 0 && otaTimer < 2 );
+
+    Posix_OtaDeleteTimer( otaTimer );
```

Comment:
Shouldn't we create a timer before deleting it?

Reply:
No. We don't require the timer to be created since the `timer_delete` function in this case is not used to delete the timer.